packages-ts/starknet/src/network.ts

```diff
@@ -1,11 +1,51 @@
 import { ChildProcess, spawn } from 'child_process'
 import axios from 'axios'
 
-export abstract class IntegratedDevnet {
+const DEVNET_NAME = 'devnet'
+const HARDHAT_NAME = 'hardhat'
+
+export class NetworkManager {
+  private opts?: any
+  private strategyDevnet: StarknetDevnet
+  private strategyHardhat: HardHatNode
+
+  constructor(opts?: any) {
+    this.opts = opts
+
+    const network = process.env.NETWORK
+    const hardhatNetwork = process.env.NETWORK_ETHEREUM
+    if (network === DEVNET_NAME) {
+      this.strategyDevnet = new StarknetDevnet('5050', this.opts)
+    }
+    if (hardhatNetwork === HARDHAT_NAME) {
+      this.strategyHardhat = new HardHatNode('8545', this.opts)
+    }
+  }
+  // This function adds some funds to pre-deployed account that we are using in our test.
+  public async start(): Promise<void> {
+    if (this.strategyDevnet) {
+      await this.strategyDevnet.startNetwork()
+    }
+    if (this.strategyHardhat) {
+      await this.strategyHardhat.startNode()
+    }
+  }
+
+  public stop() {
+    if (this.strategyDevnet) {
+      this.strategyDevnet.stop()
```

Comment:
should this set 'strategyDevnet' to null also, after stopping? Same question for `strategyHardhat`

Reply:
It should be ok to keep them like this, as we constructed them in the constructor and here we just manage their lifecycle. We should be able to start them again, after stopping, using the same manager.